src/main/java/com/conveyal/r5/analyst/cluster/AnalystWorker.java

```diff
@@ -327,84 +326,65 @@ private void handleOneRequest(GenericClusterRequest clusterRequest) {
             // Get the graph object for the ID given in the request, fetching inputs and building as needed.
             // All requests handled together are for the same graph, and this call is synchronized so the graph will
             // only be built once.
             // Record graphId so we "stick" to this same graph on subsequent polls.
             // TODO allow for a list of multiple cached TransitNetworks.
-            networkId = clusterRequest.graphId;
+            networkId = request.graphId;
             // TODO fetch the scenario-applied transportNetwork out here, maybe using OptionalResult instead of exceptions
             TransportNetwork transportNetwork = null;
             try {
                 // FIXME ideally we should just be passing the scenario object into this function, and another separate function should get the scenario object from the cluster request.
-                transportNetwork = transportNetworkCache.getNetworkForScenario(networkId, clusterRequest.extractProfileRequest());
+                transportNetwork = transportNetworkCache.getNetworkForScenario(networkId, request);
             } catch (ScenarioApplicationException scenarioException) {
                 // Handle exceptions specifically representing a failure to apply the scenario.
                 // These exceptions can be turned into structured JSON.
                 // Report the error back to the broker, which can then pass it back out to the client.
                 // Any other kinds of exceptions will be caught by the outer catch clause
-                reportTaskErrors(clusterRequest.taskId, HttpStatus.BAD_REQUEST_400, scenarioException.taskErrors);
+                reportTaskErrors(request.taskId, HttpStatus.BAD_REQUEST_400, scenarioException.taskErrors);
                 return;
             }
-            // FIXME manually coded polymorphism - we want to just combine these two request classes into one.
-            if (clusterRequest instanceof TravelTimeSurfaceRequest) {
-                this.handleTravelTimeSurfaceRequest((TravelTimeSurfaceRequest) clusterRequest, transportNetwork);
-            } else if (clusterRequest instanceof GridRequest) {
-                this.handleGridRequest((GridRequest) clusterRequest, transportNetwork);
+
+            TravelTimeComputer computer = new TravelTimeComputer(request, transportNetwork, gridCache);
+            if (request.isHighPriority()) {
+                PipedInputStream pis = new PipedInputStream();
+
+                // gzip the data before sending it to the broker. Compression ratios here are extreme (100x is not uncommon)
+                // so this will help avoid broken pipes, connection reset by peer, buffer overflows, etc. since these types
+                // of errors are more common on large files.
+                GZIPOutputStream pos = new GZIPOutputStream(new PipedOutputStream(pis));
+
+                finishPriorityTask(request, pis, "application/octet-stream", "gzip");
```

Comment:
This still strikes me as manually handled polymorphism. It's not ideal to have classes with a field that says what subtype they are (in this case AnalysisRequest.Type), then checking that subtype field (`if (request.type == X)`) and calling different functions based on the result. We should just have subtypes with different fields, and call a single method on those objects that behaves differently depending on what type they are. The finishPriorityTask method seems like something that should be on a SinglePointAnalysisRequest subclass of AnalysisRequest.